File: 121_Best_Time_To_Sell_Stock_I.java
class Solution {
    public int maxProfit(int[] prices) {
        if (prices.length == 0)
            return 0;
        
        int min_buy = prices[0], max_buy = 0;
        for (int i =0; i < prices.length; i++){
            min_buy = Math.min(min_buy, prices[i]);
            max_buy = Math.max(max_buy, prices[i] - min_buy);
        }
        return max_buy;
    }
}

File: Set1_1000/121_Best_Time_To_Sell_Stock_I.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) == 0:
            return 0
        
        min_buy = prices[0]
        max_buy = 0
        
        for elements in prices:
            min_buy = min(min_buy, elements)
            max_buy = max(max_buy, elements - min_buy)
            
        return max_buy
    
